block/src/qcow/mod.rs

```diff
@@ -2010,6 +2025,39 @@ mod tests {
         assert_eq!(read_header.backing_file_path, header.backing_file_path);
     }
 
+    #[test]
+    fn max_nesting_backing() {
+        let test_img_dir = TempDir::new_with_prefix("/tmp/ch").unwrap();
+        let path = test_img_dir.as_path().join("test.img");
+        let header = QcowHeader::create_for_size_and_path(3, 0x10_0000, path.to_str())
+            .expect("Failed to create header.");
+        let mut disk_file = RawFile::new(
+            File::create(path.as_path()).expect("Failed to create image file."),
+            false,
+        );
+        header
+            .write_to(&mut disk_file)
+            .expect("Failed to write header to file.");
+        drop(disk_file);
+        drop(header);
```

Comment:
No need to explicitly drop `header`.

If you don't rely on any side effect when dropping `disk_file`, you don't need to drop it here.

Reply:
I've wrapped it in a function for now, maybe it can be used for other test cases as well.